File: nim/day3_part1_2016.nim
import strutils, sequtils

proc isPossibleTriangle(a, b, c: int): bool =
  (a + b > c) and (a + c > b) and (b + c > a)

proc countPossibleTriangles(filename: string): int =
  var count = 0
  for line in lines(filename):
    let sides = line.splitWhitespace().mapIt(it.parseInt())
    if isPossibleTriangle(sides[0], sides[1], sides[2]):
      inc count
  return count

echo countPossibleTriangles("input.txt")

File: training_data/reflection-tuning/day3_part1_2016.clj
(defn valid-triangle? [[a b c]]
  (and (> (+ a b) c)
       (> (+ b c) a)
       (> (+ c a) b)))

(defn parse-line [line]
  (map #(Integer/parseInt %) (re-seq #"\d+" line)))

(defn count-valid-triangles [filename]
  (->> (slurp filename)
       clojure.string/split-lines
       (map parse-line)
       (filter valid-triangle?)
       count))

(println (count-valid-triangles "input.txt"))
